### PR TITLE
Adding pinot file system command

### PR DESCRIPTION
**Pull request by xiangfu0**
Adding Pinot FS command into toolings for quickly testing if the filesystem configs and settings credentials are correct. Also, users can perform some simple actions if needed.

Usage:
```
pinot-admins.sh FileSystem <sub-command> <sub-command parameters>
```
There is also a separate script for it:
```
pinot-fs-util.sh <sub-command> <sub-command parameters>
```

Currently, supported sub-command operations are coming from PinotFS interfaces:
```
ls
cp
mv
rm
```

Default loaded PinotFS are `org.apache.pinot.spi.filesystem.LocalPinotFS` and `org.apache.pinot.plugin.filesystem.S3PinotFS` if scheme is not registered by the config file.

Example:

List files recursively on s3 when AWS is configured:
```
pinot-admin.sh FileSystem ls -r s3://<my-bucket>/
```

List files recursively on s3 with the pinot fs config:
```
pinot-admin.sh  FileSystem -config /tmp/fs.conf  -command ls -r s3://<my-bucket>/
```

The content of `fs.conf` will be deserailize to `org.apache.pinot.spi.ingestion.batch.spec.PinotFSSpec`
```
class.s3=org.apache.pinot.plugin.filesystem.S3PinotFS
s3.region=us-west-2
s3.accessKey=<my-access-key>
s3.secretKey=<my-secret-key>
```

Examples of no access for s3:
```
bin/pinot-admin.sh  FileSystem ls  s3://<my-bucket>

Unable to list files under: s3://<my-bucket>, exception: software.amazon.awssdk.services.s3.model.S3Exception: Access Denied (Service: S3, Status Code: 403, Request ID: 33KA59XA0ZPBDPN2, Extended Request ID: TILm5sJyqoQttVTNgSS5Hs1N89cQa9MtxRAh5oQSH8Jm9DpOG/yI/raYIw/7FU3fYzARba6qQBA=)
```

### CODE REVIEW

File: pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/FileSystemCommand.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+          break;
+        case "MV":
+        case "MOVE":
+          output = String.valueOf(pinotFS.move(URI.create(arguments[0]), URI.create(arguments[1]), true));
+          break;
+        case "MKDIR":
+          output = String.valueOf(pinotFS.mkdir(URI.create(arguments[0])));
+          break;
+        default:
+          output = "Unsupported file system op: " + fsOp + ". Current supported operations are: "
```
Comment (Jackie-Jiang) [inline pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/FileSystemCommand.java:186]: We should throw exception, or user might think the operation is supported

File: pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/FileSystemCommand.java [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+              + StringUtil.join(", ", SUPPORTED_FS_COMMANDS);
+          break;
+      }
+    } catch (Exception e) {
+      e.printStackTrace(new PrintStream(output));
+    }
+    return output;
+  }
+
+  private void registerDefaultPinotFS() {
```
[the remark this answers is not in the document]
Comment (xiangfu0) [reply pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/FileSystemCommand.java:196]: This is part of pinot-tools module and has s3 packaged already.

File: pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/FileSystemCommand.java [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+import picocli.CommandLine;
+
+
+@CommandLine.Command(name = "FileSystem")
+public class FileSystemCommand extends AbstractBaseAdminCommand implements Command {
+  private static final Logger LOGGER = LoggerFactory.getLogger(FileSystemCommand.class.getName());
+
+  private static final String[] SUPPORTED_FS_COMMANDS = new String[]{
+      "EXISTS", "IS_DIR", "LAST_MODIFIED", "LENGTH",
+      "LIST_FILE", "LIST_FILE_RECURSIVE",
```
[the remark this answers is not in the document]
Comment (xiangfu0) [reply pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/FileSystemCommand.java:46]: PinotFS has no LS support, only List file, not able to list directory 😮‍💨  
Or I can add a new interface in PinotFS for `list` and `list_directory`

[the remark this answers is not in the document]
Comment (Jackie-Jiang) [reply pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/FileSystemCommand.java:46]: Current `listFiles()` should list all the files and directories under the given path, which is the same as `ls` right?

File: pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/FileSystemCommand.java [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+  @Override
+  public String getName() {
+    return "FileSystem";
+  }
+
+  @Override
+  public String toString() {
+    return ("FileSystem --scheme " + _scheme + " -configFileName " + _configFile + " -command " + Arrays.toString(
+        _command));
```
[the remark this answers is not in the document]
Comment (xiangfu0) [reply pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/FileSystemCommand.java]: Moved to subcommand, looks much cleaner.

File: pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/filesystem/ListFiles.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+
+@CommandLine.Command(name = "ls", aliases = {"list"}, description = "List files")
+public class ListFiles extends BaseFileOperation {
+  @CommandLine.Option(names = {"-r", "--recursive"},
+      description = "Recursively list subdirectories")
+  private boolean _recursive;
+
+  @CommandLine.Parameters(arity = "1..*", description = "File paths to list")
+  private String[] _filePaths;
```
Comment (walterddr) [inline pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/filesystem/ListFiles.java:35]: do we really want ppl to list multiple paths?

**Review by walterddr — APPROVED** [review]

File: pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/filesystem/ListFiles.java (hunk 5 above)
Comment (xiangfu0) [reply pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/filesystem/ListFiles.java:35]: seems ls supports multiple parameters

**Pull request merged**